tests/integration_test/core_mqtt_system_test.c

```diff
@@ -770,7 +770,7 @@ static void startPersistentSession()
     TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_SUCCESS, SecureSocketsTransport_Connect( &networkContext,
                                                                                         &serverInfo,
                                                                                         &socketsConfig ) );
-    TEST_ASSERT_NOT_NULL( networkContext.pContext );
+    TEST_ASSERT_NOT_NULL( networkContext.tcpSocket );
```

Comment:
I removed this `NULL` check asserts from the MQTT integration test in PR #2514 as only checking the return value of `SecureSocketsTransport_Connect` should suffice in the test. The Secure Sockets based implementation of the transport interface is encapsulated in the `transport_secure_sockets.c` implementation, and this test shouldn't need to test the correctness of the `SecureSocketsTransport_Connect` API.
(More importantly, a NULL-check against the socket cannot be performed as some vendor implementations of `iot_secure_sockets.h` treat `Socket_t` type as an integer.)

Reply:
removed null check in de398277dbb6eede0480df8a1a53b7a6515ca4b6